### ISSUE TITLE
Choosing the same file in expo-document-picker crashes iOS simulator

### ISSUE THREAD

**Issue by geoRG77** [expo/expo#16096]
### Summary

I'm facing an issue with `expo-document-picker` which is affecting only iOS simulator, on real device, there is no issue. When I select a file and then select the same file one more time, the app immediately crashes.

I have tried:
- to use a different iOS version (15.2, 15.0, 14.5) => no change
- to use the latest `expo-document-picker@10.1.1` from yesterday but it seems to be present in all `10.x.x` versions
- to use `expo prebuild` and debug the issue in Xcode, I saw `EXC_BAD_ACCESS` in the native `EXDocumentPickerModule.m` once it crashed but I don't have enough knowledge to fix it

Thank you for any help, I'm trying to solve it for a few days with no luck.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

SDK 44

### Environment

```
Expo CLI 5.0.3 environment info:
    System:
      OS: macOS 12.2
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
      Yarn: 1.22.15 - ~/.nvm/versions/node/v16.13.1/bin/yarn
      npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    IDEs:
      Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~44.0.2 => 44.0.5
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.1 => 0.17.1
    npmGlobalPackages:
      eas-cli: 0.45.1
      expo-cli: 5.0.3
    Expo Workflow: bare
```

### Reproducible demo

```javascript
export default function App() {
  let selectDocument = async () => {
    let file = await DocumentPicker.getDocumentAsync({});
    console.log(file.name);
  };

  return (
    <View style={styles.container}>
      <StatusBar style="auto" />
      <Button title="Upload file" onPress={selectDocument} />
    </View>
  );
}
```

Snack: https://snack.expo.dev/@jiri.orsag/document-picker-issue

**Comment by brentvatne** [comment]
hi there! i cannot reproduce this on the snack or by copying that exact code from the snack and running it locally. are you able to reproduce it on the snack that you provided?

**Comment by geoRG77** [comment]
Hi @brentvatne, thanks for checking! Sometimes it works using the snack for me too 😵 But I prepared full repo where the issue is always happening: https://github.com/geoRG77/document-picker-issue

**Comment by geoRG77** [comment]
Adding short screen recording
https://user-images.githubusercontent.com/735057/151519849-68e37bed-70ed-4034-98e7-d8322c9b7ef9.mov

**Comment by taschdo** [comment]
I have the same problem. Moreover, everything worked in SDK 42. Share the solution who will find it.

**Comment by brentvatne** [comment]
@geoRG77 - what file format is that sample file?

I'm able to reproduce this when selecting a keynote file - in this case the issue is that the returned mime type is nil and that results in a crash (too bad we didn't write this in Swift :P)

**Comment by geoRG77** [comment]
> @geoRG77 - what file format is that sample file?
> 
> I'm able to reproduce this when selecting a keynote file - in this case the issue is that the returned mime type is nil and that results in a crash (too bad we didn't write this in Swift :P)

@brentvatne It's a PDF, I tried a few of them but it didn't matter what type it was. Originally I thought it might be crashing because of some obscure PDF format. I see the PR, thank you 👍

**Comment by NotifydLLC** [comment]
We have the same issue, and it appears to be occurring for any file type.

**Comment by brentvatne** [comment]
@geoRG77 @NotifydLLC - can you try applying the change from https://github.com/expo/expo/pull/16156 to your node_modules with [patch-package](https://www.npmjs.com/package/patch-package), then building with EAS Build (or `expo run:ios`) and verifying if this fixes the issue for you?

**Comment by geoRG77** [comment]
@brentvatne I changed the file, created the patch but no luck with either `expo run:ios`, or `expo start`

**Comment by brentvatne** [comment]
@geoRG77 - can you try expo-document-picker@10.1.3? this includes the fix from #16156

**Comment by geoRG77** [comment]
@brentvatne I did and it works! Actually, it works even in Expo Go. Thank you 👍
